File: 백준/Bronze/2231. 분해합/분해합.py
num = int(input())
arr=[]
for i in range(num):
    sum=0
    for j in str(i):
        sum+=int(j)
    sum+=i
    if sum == num:
        arr.append(i)
if len(arr)>0:
    print(min(arr))
else:
    print(0)

File: 백준/Bronze/2231. 분해합/분해합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        for (int i = 1; i<=n; i++) {
            String num = String.valueOf(i);
            int sum = i;
            for(int j=0; j<num.length(); j++) {
                sum+=(Integer.parseInt(String.valueOf(num.charAt(j))));
            }

            if(sum == n) {
                System.out.println(i);
                return;
            }
        }

        System.out.println(0);

    }
}